cpp/arcticdb/arrow/arrow_schema.hpp

```diff
@@ -0,0 +1,289 @@
+#pragma once
+
+#include <arcticdb/arrow/arrow_c_data_interface.hpp>
+#include <arcticdb/arrow/arrow_wrappers.h>
+#include <arcticdb/arrow/nanoarrow/nanoarrow.h>
+#include <arcticdb/arrow/arrow_error.hpp>
+#include <arcticdb/util/preconditions.hpp>
+
```

Comment:
`std::optional` is unresolved with gcc 12.3.0-3.
```suggestion

#include <optional>
```